barbershop/worker_profile/views.py

```diff
@@ -1,3 +1,42 @@
 from django.shortcuts import render
+from worker_profile.models import WorkerProfile, WorkerCommunications
+from worker_profile.serializers import WorkerProfileSerializer, WorkerCommunicationsSerializer
+from rest_framework import generics
+from rest_framework.pagination import LimitOffsetPagination
+from rest_framework.permissions import IsAdminUser
+from django.shortcuts import get_object_or_404
+
 
 # Create your views here.
+
+
+class WorkerProfileList(generics.ListCreateAPIView):
+    queryset = WorkerProfile.objects.all()
+    serializer_class = WorkerProfileSerializer
+    permission_classes = [IsAdminUser]
```

Comment:
rework please / Use self-written perm class that should check if user is in the group "workers"(or any other)